api/src/main/java/com/ford/labs/retroquest/metrics/Metrics.java

```diff
@@ -25,54 +19,59 @@ public Metrics(TeamRepository teamRepository, FeedbackRepository feedbackReposit
+        return getFeedbackCount(null, null);
     }
 
-    public int getFeedbackCount(LocalDate startTime, LocalDate endTime) {
-        LocalDateTime finalEndTime = endTime == null ? LocalDateTime.now() : endTime.atStartOfDay();
-        LocalDateTime finalStartTime = startTime == null ? LocalDateTime.MIN : startTime.atStartOfDay();
-        List<Feedback> feedbackList = feedbackRepository.findAll().stream()
-                .filter(feedback -> !feedback.getDateCreated().isBefore(finalStartTime))
-                .filter(feedback -> !feedback.getDateCreated().isAfter(finalEndTime))
-                .collect(toList());
-        return feedbackList.size();
+    public int getFeedbackCount(LocalDate startDate, LocalDate endDate) {
+        var dateRange = DateTimeRange.fromStartAndEnd(startDate, endDate);
+
+        return (int) feedbackRepository
+            .countByDateCreatedGreaterThanEqualAndDateCreatedLessThanEqual(
+                dateRange.getStartDateTime(),
+                dateRange.getEndDateTime()
+            );
     }
 
     @ManagedAttribute
     public double getAverageRating() {
-        List<Feedback> allFeedback = feedbackRepository.findAllByStarsIsGreaterThanEqual(1);
-        OptionalDouble average = allFeedback.stream().mapToDouble(Feedback::getStars).average();
-        return average.isPresent() ? average.getAsDouble() : 0.0;
+        return getAverageRating(null, null);
     }
 
-    public double getAverageRating(LocalDate startTime, LocalDate endTime) {
-        LocalDateTime finalEndTime = endTime == null ? LocalDateTime.now() : endTime.atStartOfDay();
-        LocalDateTime finalStartTime = startTime == null ? LocalDateTime.MIN : startTime.atStartOfDay();
-        return feedbackRepository.findAllByStarsIsGreaterThanEqual(1).stream()
-                .filter(feedback -> !feedback.getDateCreated().isBefore(finalStartTime))
-                .filter(feedback -> !feedback.getDateCreated().isAfter(finalEndTime))
-                .mapToInt(Feedback::getStars)
-                .average()
-                .orElse(0);
+    public double getAverageRating(LocalDate startDate, LocalDate endDate) {
+        var dateRange = DateTimeRange.fromStartAndEnd(startDate, endDate);
+
+        return feedbackRepository
+            .getAverageRating(
+                dateRange.getStartDateTime(),
+                dateRange.getEndDateTime()
+            );
+    }
+
+    public int getTeamLogins() {
+        return getTeamLogins(null, null);
     }
 
     public int getTeamLogins(LocalDate startDate, LocalDate endDate) {
-        LocalDate finalEndDate = endDate == null ? LocalDate.now() : endDate;
-        LocalDate finalStartDate = startDate == null ? LocalDate.of(1900, 1, 1) : startDate;
-        return teamRepository.findAllByLastLoginDateBetween(finalStartDate, finalEndDate).size();
+        var dateRange = DateTimeRange.fromStartAndEnd(startDate, endDate);
+
+        return (int) teamRepository.countByLastLoginDateBetween(
```

Comment:
pass objects in repository query use CustomQuery to create

https://spring.io/blog/2014/07/15/spel-support-in-spring-data-jpa-query-definitions